sofiAPI.py

```diff
@@ -0,0 +1,634 @@
+
+            if orderObj.get_action() == "buy":
+                clicked_values = set()  # Set to keep track of processed accounts
+
+                DRY = orderObj.get_dry()
+                QUANTITY = orderObj.get_amount()
+                print("DRY MODE:", DRY)
+                account_number = 1
+                sleep(4)
+
+                while True:
+                    sleep(1)
+                    try:
+                        print("Attempting to click the buy button")
+                        buy_button = WebDriverWait(driver, 20).until(
+                            EC.element_to_be_clickable((By.XPATH, "//*[@id='mainContent']/div[2]/div[2]/div[2]/div[2]/div/button[1]")))
+                        driver.execute_script("arguments[0].click();", buy_button)
+                        print("Buy button clicked")
+                    except TimeoutException:
+                        print(f"Buy button not found for {s}. Moving to next stock.")
+                        printAndDiscord(f"SOFI buy button not found for {s}.", loop)
+                        break
+
+                    try:
+                        dropdown = WebDriverWait(driver, 5).until(
+                            EC.element_to_be_clickable((By.XPATH, "//*[@id='OrderTypedDropDown']")))
+                        selected_option = dropdown.find_element(By.XPATH, '//*[@id="mainContent"]/div[2]/div[2]/div[3]/div/div[3]/p[2]').text
+                        if selected_option != "Limit price":
+                            OrderTypedDropDown_limit = WebDriverWait(driver, 3).until(
+                                EC.element_to_be_clickable((By.XPATH, "//*[@id='OrderTypedDropDown']/option[2]")))
+                            OrderTypedDropDown_limit.click()
+                    except (NoSuchElementException, TimeoutException):
+                        pass
+
+                    accounts_dropdown = WebDriverWait(driver, 20).until(
+                        EC.element_to_be_clickable((By.NAME, "account")))
+                    select = Select(accounts_dropdown)
+
+                    for option in select.options:
+                        value = option.get_attribute('value')
+                        if value not in clicked_values:
+                            select.select_by_value(value)
+                            print(f"Selected account {account_number} (real: {key}): {value} (dropdown selection value)")
+                            account_number += 1
+                            clicked_values.add(value)
+                            break
+                    else:
+                        print(f"All accounts have been processed for {s}.")
+                        break
+
+                    try:
+                        print("Fetching live price for the stock")
+                        live_price_element = WebDriverWait(driver, 20).until(
+                            EC.presence_of_element_located((By.XPATH, "//*[@id='mainContent']/div[2]/div[2]/div[3]/div/p")))
```

Comment:
```python
live_price_element = WebDriverWait(driver, 20).until(EC.presence_of_element_located((By.XPATH, "//*[@id='mainContent']/div[2]/div[2]/div[3]/div/p")))
```

This on line 460 is not showing up for me on Edge or Firefox (untested on standard Google Chrome) no matter what I set my monitor resolution to. The market price div Xpath is `//*[@id="mainContent"]/div[2]/div[2]/div[3]/div/div[4]/div[2]` if that is what this represents. Otherwise, is this something that only shows when the market is open?

Reply:
looks like its one of those things that its being thrown off by if theres a current number of shares of that stock or not element, completely missed it.